neo/Network/P2P/RemoteNode.cs

```diff
@@ -32,11 +32,11 @@ internal class Relay { public IInventory Inventory; }
         public uint LastBlockIndex { get; private set; } = 0;
         public bool IsFullNode { get; private set; } = false;
 
-        public RemoteNode(NeoSystem system, object connection, IPEndPoint remote, IPEndPoint local)
+        public RemoteNode(NeoSystem system, object connection, IPEndPoint remote, IPEndPoint local, IActorRef protocolHandler)
```

Comment:
Can we remove the `protocolHandler` parameter?

Reply:
Perhaps, but we need to think... on tests, it is created like:
```cs
var protocolActor = ActorOfAsTestActorRef<ProtocolHandler>(() => new ProtocolHandler(testBlockchain));
```
But if automated on Neo project, it will be created like: 
```cs
var protocol = system.ActorSystem.ActorOf(ProtocolHandler.Props(system));
```
This would require a Factory, as Ricardo did... but I think it's less complicated to just receive the protocolHandler as parameter.

Reply:
One possibility is to put some actor factory method on NeoSystem, so we require it to create the actors. This would solve both problems I think.

Reply:
What's the difference when it is created with `ActorOfAsTestActorRef()`?

Reply:
The only difference is that it returns a TestActorRef instead of an ActorRef. This allows it to be tested under TestKit Sys environment, which I guess, bypasses parts of a global akka ActorSystem. I think the key is to allow testing under single-threaded independent tests, even if tests are run simultaneously.

Reply:
Hello,

I got this from the [docs](http://api.getakka.net/docs/stable/html/120DDADF.htm):
"This special ActorRef is exclusively for use during unit testing in a single-threaded environment. Therefore, it overrides the dispatcher to CallingThreadDispatcher and sets the receiveTimeout to None. Otherwise, it acts just like a normal ActorRef. You may retrieve a reference to the underlying actor to test internal logic. A TestActorRef< TActor> can be implicitly casted to an IActorRef or you can get the actual IActorRef from the Ref property."

I believe another alternative would be to use dependency injection. Somehow we must be able to test these classes and mock their behavior using Akka system. I'm not an expert in Akka, so I just did what was "suggested".

Reply:
Parameter removed Erik, managed to make it work without any other big change.